static/js/ui_init.js

```diff
@@ -231,6 +231,17 @@ $(function () {
         $(this).tooltip('hide');
     });
 
+    // Refocus the compose area after clicking links
+    $("body").on("focus", '*', function (e) {
+        if (e.target.id === 'new_message_content') {
+            $(".messagebox-content a").off("click.refocus").on("click.refocus", function () {
+                $("#new_message_content").focus();
+            });
+        } else if (e.target.tagName !== 'A') {
+            $(".messagebox-content a").off("click.refocus");
+        }
+    });
```

Comment:
@timabbott  Thank you for your advice. I totally understand your worry. There is already a commit before preventing focus to going to the link. I just want to make sure the message box will be only refocused just after itself defocused, not other input boxes. So instead of remove "refocus" handler when a user clicks other input boxes, I think it is possible to use a variable to record the last focused input box and check it when we click links in the message box. Do you think it is a possible way?